scripts/jenkins/groovy/insideDocker.groovy

```diff
@@ -19,7 +19,14 @@ def call(customEnv, image, registry, buildConfig, timeoutValue, timeoutUnit, cus
     timeout(time: timeoutValue, unit: timeoutUnit) {
       docker.withRegistry("https://${registry}") {
         withCredentials([file(credentialsId: 'c096a055-bb45-4dac-ba5e-10e6e470f37e', variable: 'JUNIT_CORE_SITE_PATH'), [$class: 'AmazonWebServicesCredentialsBinding', accessKeyVariable: 'AWS_ACCESS_KEY_ID', credentialsId: 'AWS S3 Credentials', secretKeyVariable: 'AWS_SECRET_ACCESS_KEY']]) {
-          docker.image(image).inside("--init --add-host=nexus:172.17.0.53 --dns ${DEFAULT_DNS} -e AWS_ACCESS_KEY_ID=\${AWS_ACCESS_KEY_ID} -e AWS_SECRET_ACCESS_KEY=\${AWS_SECRET_ACCESS_KEY} -v /home/0xdiag:/home/0xdiag -v /home/jenkins/repos:/home/jenkins/repos ${customArgs}") {
+          def groupAddArg = ""
+          dockerGroupName = "docker"
```

Comment:
what about making this an optional arg and only if its specified run the code below (only for k8s stage)
or to do this somehow via extraDockerArgs or some simmilar option in defineTestStages

Reply:
Unless we want monstrous changes in the pipeline, the if would have to be placed here, as Michal Raska told above. The GID is known here, not in the earlier phase. Let's control this via the boolean. Looks like good compromise.